### PR TITLE
Remove OAuth authentication for GitHub

### PR DESCRIPTION
**Pull request by nedbat**
Also, improve the error handling to be sure we can see what's happening when an exception occurs.

### CODE REVIEW

**Comment by nedbat** [comment]
@Carlos-Muniz @kdmccormick These changes make the bot work well for the openedx organization.

**Review by kdmccormick — APPROVED** [review]

**Review by kdmccormick — APPROVED** [review]

File: openedx_webhooks/tasks/github.py [hunk 1]
```diff
@@ -29,7 +30,13 @@
...
 def pull_request_changed_task(_, pull_request):
     """A bound Celery task to call pull_request_changed."""
-    return pull_request_changed(pull_request)
+    try:
+        ret = pull_request_changed(pull_request)
+        log_rate_limit()
+    except Exception as exc:
+        logger.exception("Couldn't pull_request_changed_task")
+        raise
+
```
Comment (Carlos-Muniz) [inline openedx_webhooks/tasks/github.py:39]: Missing coverage

File: openedx_webhooks/utils.py [hunk 2]
```diff
@@ -76,7 +76,19 @@ def log_check_response(response, raise_for_status=True):
     msg = "Response: {0.status_code} {0.reason!r} for {0.url}: {0.content!r}".format(response)
     logger.debug(msg)
     if raise_for_status:
-        response.raise_for_status()
+        try:
+            response.raise_for_status()
+        except Exception as exc:
+            req = response.request
+            logger.exception(f"HTTP request failed: {req.method} {req.url}. Response body: {response.content}")
+            raise
```
Comment (Carlos-Muniz) [inline openedx_webhooks/utils.py:84]: Missing coverage

File: openedx_webhooks/utils.py [hunk 3]
```diff
@@ -76,7 +76,19 @@ def log_check_response(response, raise_for_status=True):
...
+
+
+def log_rate_limit():
+    """Get stats from GitHub about the current rate limit, and log them."""
+    rate = get_github_session().get("/rate_limit").json()['rate']
+    reset = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(rate['reset']))
+    logger.info(f"Rate limit: {rate['limit']}, used {rate['used']}, remaining {rate['remaining']}. Reset is at {reset}")
```
Comment (Carlos-Muniz) [inline openedx_webhooks/utils.py:91]: Missing coverage

File: openedx_webhooks/utils.py [hunk 4]
```diff
@@ -156,17 +168,10 @@ def paginated_get(url, session=None, limit=None, per_page=100, callback=None, **
     returned = 0
     while url:
         resp = retry_get(session, url, **kwargs)
+        log_check_response(resp)
         if callable(callback):
             callback(resp)
```
Comment (Carlos-Muniz) [inline openedx_webhooks/utils.py:173]: Missing Coverage

**Review by Carlos-Muniz — COMMENTED** [review]
There is some coverage missing. Can tests be written for these?

File: openedx_webhooks/tasks/github.py (hunk 1 above)
Comment (nedbat) [reply openedx_webhooks/tasks/github.py:39]: Oh snap! :)

**Pull request merged**